lib/Conversion/SimToSV/SimToSV.cpp

```diff
@@ -0,0 +1,138 @@
+                  ConversionPatternRewriter &rewriter) const final {
+    auto loc = op.getLoc();
+
+    auto i1ty = rewriter.getIntegerType(1);
+    auto type = op.getResult().getType();
+
+    auto regv = rewriter.create<sv::RegOp>(loc, type,
+                                           rewriter.getStringAttr("_pargs_v_"));
+    auto regf = rewriter.create<sv::RegOp>(loc, i1ty,
+                                           rewriter.getStringAttr("_pargs_f"));
+
+    rewriter.create<sv::IfDefOp>(
+        loc, "SYNTHESIS",
+        [&]() {
+          auto cstFalse = rewriter.create<hw::ConstantOp>(loc, APInt(1, 0));
+          auto cstZ = rewriter.create<sv::ConstantZOp>(loc, type);
+          auto assignZ = rewriter.create<sv::AssignOp>(loc, regv, cstZ);
+          circt::sv::setSVAttributes(
+              assignZ,
+              sv::SVAttributeAttr::get(
+                  rewriter.getContext(),
+                  "This dummy assignment exists to avoid undriven lint "
+                  "warnings (e.g., Verilator UNDRIVEN).",
+                  /*emitAsComment=*/true));
+          rewriter.create<sv::AssignOp>(loc, regf, cstFalse);
+        },
+        [&]() {
+          rewriter.create<sv::InitialOp>(loc, [&] {
+            auto zero32 = rewriter.create<hw::ConstantOp>(loc, APInt(32, 0));
+            auto tmpResultType = rewriter.getIntegerType(32);
+            auto str =
+                rewriter.create<sv::ConstantStrOp>(loc, op.getFormatString());
+            auto call = rewriter.create<sv::SystemFunctionOp>(
+                loc, tmpResultType, "value$plusargs",
+                ArrayRef<Value>{str, regv});
+            auto test = rewriter.create<comb::ICmpOp>(
+                loc, comb::ICmpPredicate::ne, call, zero32, true);
+            rewriter.create<sv::BPAssignOp>(loc, regf, test);
+          });
+        });
+
+    auto readf = rewriter.create<sv::ReadInOutOp>(loc, regf);
+    auto readv = rewriter.create<sv::ReadInOutOp>(loc, regv);
+    rewriter.replaceOp(op, {readf, readv});
+    return success();
+  }
+};
+
+namespace {
+#define GEN_PASS_DEF_LOWERSIMTOSV
+#include "circt/Conversion/Passes.h.inc"
+
+struct SimToSVPass : public impl::LowerSimToSVBase<SimToSVPass> {
+  void runOnOperation() override {
+    auto circuit = getOperation();
+    MLIRContext *context = &getContext();
+
+    ConversionTarget target(*context);
+    target.addIllegalDialect<SimDialect>();
+    target.addLegalDialect<sv::SVDialect>();
+    target.addLegalDialect<hw::HWDialect>();
+    target.addLegalDialect<comb::CombDialect>();
+
+    RewritePatternSet patterns(context);
+    patterns.add<PlusArgsTestLowering>(context);
+    patterns.add<PlusArgsValueLowering>(context);
+    if (failed(applyPartialConversion(circuit, target, std::move(patterns))))
+      signalPassFailure();
+  }
+};
+} // anonymous namespace
+
+std::unique_ptr<Pass> circt::createLowerSimToSVPass() {
+  return std::make_unique<SimToSVPass>();
+}
```

Comment:
I did not review this file in detail as this looked like it was (correctly) copying over logic from `LowerToHW`. Verify that this is definitely not adding anything new.